Pages/2-basics.md

```diff
@@ -83,57 +143,226 @@ Variables are created with `def`
 (def weight 5)
 ```
 
+These get defined in the namespace of the file they are defined in.
+
+Note that these are not the customary "bash in place" variables you might be used to.  We are simply **binding** a **symbol** to a **value** here, and making that available in the namespace we are working in.
+
+
 ### Functions
+Functions - those formerly mystical **values** that can be **evaluated** to produce **values** from other **values** - where do they come from?
 
-Here's an anonymous function: 
+Why, there is a form to create them, of course.
 
-```
-((fn [name] (clojure.string/upper-case name)) "bob")
+```Clojure
+(fn [name]
+   (clojure.string/upper-case name))
+   
 ```
 
-Oh boy, lots of parens! Lets break it down, starting in the innermost expression:
+Evaluating this form produces a function (a value) which does what it says.  It **binds** name to the argument passed, and then uses that to call another function ('upper-case', which is in the namespace 'clojure.string') and returns that value.
 
-```
-(clojure.string/upper-case name)
-```
+This one is anonymous - it went *poof* right after it became a value.  We had the value, but the function was never used.  Not so useful, let's do something different - 
```

Comment:
hm the example i had before had a value, and you make it sound like they are "never" useful. You can mention "Not useful alone, but you might write an anonymous function as parameter to another function.